torch/nn/attention/_flex_attention.py

```diff
@@ -231,65 +231,72 @@ def _create_block_mask_from_mask(
 def _create_mask(
     score_mod: _score_mod_signature,
     B: int,
-    H: int,
+    Hkv: int,
```

Comment:
Why make it another dimension?

Reply:
score_mod takes (b, hq, m, n) but the block mask is of shape (B, Hkv, (Hq//Hkv)*M, N) to matches our internal shape. We need the group size information to properly cast the hkv to hq. Maybe we should take a tuple of `(Hkv, Hq)` instead?

Reply:
Can we not hide this from the user?

Reply:
Hide this detail from user: broadcast mask along G dimension instead `_flex_attention` via `_broadcast_block_mask_for_gqa`.